packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx

```diff
@@ -0,0 +1,191 @@
+// @ts-strict-ignore
+import React, { useState } from 'react';
+
+import { send } from 'loot-core/src/platform/client/fetch';
+
+import { styles } from '../../../style';
+import { Block } from '../../common/Block';
+import { Menu } from '../../common/Menu';
+import { MenuButton } from '../../common/MenuButton';
+import { MenuTooltip } from '../../common/MenuTooltip';
+import { View } from '../../common/View';
+import { ChooseGraph } from '../ChooseGraph';
+import { DateRange } from '../DateRange';
+import { LoadingIndicator } from '../LoadingIndicator';
+import { ReportCard } from '../ReportCard';
+
+type CardMenuProps = {
+  onClose: () => void;
+  onMenuSelect: (item: string, reportId: string) => void;
+  reportId: string;
+};
+
+function CardMenu({ onClose, onMenuSelect, reportId }: CardMenuProps) {
+  return (
+    <MenuTooltip onClose={onClose} width={120}>
+      <Menu
+        onMenuSelect={item => {
+          onMenuSelect(item, reportId);
+        }}
+        items={[
+          {
+            name: 'rename',
+            text: 'Rename report',
+            disabled: true,
+          },
+          {
+            name: 'delete',
+            text: 'Delete report',
+          },
+        ]}
+      />
+    </MenuTooltip>
+  );
+}
+
+function index(data) {
+  const result = {};
+  data.forEach(report => {
+    result[report.id] = false;
+  });
+  return result;
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏using `reduce` would result in slightly cleaner code.
```suggestion
function index<T>(data: T[]) {
  return data.reduce((carry, report) => ({ ...carry, [report.id]: false }), []);
}
```